python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -165,36 +166,41 @@ def _makeRecords(dataIds: Mapping[str, Iterable], universe: DimensionUniverse) -
         `~lsst.daf.butler.DimensionRecord` for each input name. Related
         dimensions (e.g., instruments and detectors) are linked arbitrarily.
     """
-    expandedIds = {}
-    # Provide alternate keys like detector names
-    for name, values in dataIds.items():
-        expandedIds[name] = []
+
+    # Create values for all dimensions that are (recursive) required or implied
+    # dependencies of the given ones.
+    complete_data_id_values = {}
+    for dimension in universe.extract(dataIds.keys()):
+        if dimension.name in dataIds:
+            complete_data_id_values[dimension.name] = list(dataIds[dimension.name])
+        if dimension.name not in complete_data_id_values:
+            complete_data_id_values[dimension.name] = [_makeRandomDataIdValue(dimension)]
```

Comment:
It looks like there just are no tests of `dataIds` in daf_butler; the ones here all use `addDataIdValue` instead, which fits much better with the structure of the test, since it tries to use only one repo created in `setUpClass` (for performance reasons).  There are tests in downstream packages that use `dataIds`.